cmd/ci-secret-generator/main.go

```diff
@@ -127,55 +153,104 @@ func executeCommand(command string) ([]byte, error) {
 
+func replaceParameter(paramName, param, template string) string {
+	return strings.ReplaceAll(template, fmt.Sprintf("$(%s)", paramName), param)
+}
+
 func updateSecrets(bwItems []bitWardenItem, bwClient bitwarden.Client) error {
-	for _, bwItem := range bwItems {
-		logger := logrus.WithField("item", bwItem.ItemName)
-		if bwItem.Field.Name != "" {
-			logger = logger.WithFields(logrus.Fields{
-				"field":   bwItem.Field.Name,
-				"command": bwItem.Field.Cmd,
-			})
-			logger.Info("processing field")
-			out, err := executeCommand(bwItem.Field.Cmd)
-			if err != nil {
-				logrus.WithError(err).Error("failed to generate field")
-				return err
-			}
-			if err := bwClient.SetFieldOnItem(bwItem.ItemName, bwItem.Field.Name, out); err != nil {
-				logrus.WithError(err).Error("failed to upload field")
-				return err
+	for _, bwItemWithParams := range bwItems {
+		bwItemsQ := []bitWardenItem{bwItemWithParams}
+		for paramName, params := range bwItemWithParams.Params {
+			bwItemsProcessed := []bitWardenItem{}
+			for _, qItem := range bwItemsQ {
+				for _, param := range params {
+					argItem := bitWardenItem{}
+					err := deepcopy.Copy(&argItem, &qItem)
+					if err != nil {
+						logrus.Fatal(err)
+						return fmt.Errorf("error copying bitWardenItem: %w", err)
+					}
+					argItem.ItemName = replaceParameter(paramName, param, argItem.ItemName)
+					for i, field := range argItem.Fields {
+						argItem.Fields[i].Name = replaceParameter(paramName, param, field.Name)
+						argItem.Fields[i].Cmd = replaceParameter(paramName, param, field.Cmd)
+					}
+					for i, attachment := range argItem.Attachments {
+						argItem.Attachments[i].Name = replaceParameter(paramName, param, attachment.Name)
+						argItem.Attachments[i].Cmd = replaceParameter(paramName, param, attachment.Cmd)
+					}
+					argItem.Attribute.Name = replaceParameter(paramName, param, argItem.Attribute.Name)
+					argItem.Attribute.Cmd = replaceParameter(paramName, param, argItem.Attribute.Cmd)
+					argItem.Notes = replaceParameter(paramName, param, argItem.Notes)
+					bwItemsProcessed = append(bwItemsProcessed, argItem)
+				}
 			}
+			bwItemsQ = bwItemsProcessed
 		}
-		if bwItem.Attachment.Name != "" {
-			logger = logger.WithFields(logrus.Fields{
-				"attachment": bwItem.Attachment.Name,
-				"command":    bwItem.Attachment.Cmd,
-			})
-			logger.Info("processing attachment")
-			out, err := executeCommand(bwItem.Attachment.Cmd)
-			if err != nil {
-				logrus.WithError(err).Error("failed to generate attachment")
-				return err
+		for _, bwItem := range bwItemsQ {
+			logger := logrus.WithField("item", bwItem.ItemName)
+			for _, field := range bwItem.Fields {
+				if field.Name != "" {
+					logger = logger.WithFields(logrus.Fields{
+						"field":   field.Name,
+						"command": field.Cmd,
+					})
+					logger.Info("processing field")
+					out, err := executeCommand(field.Cmd)
+					if err != nil {
+						logrus.WithError(err).Error("failed to generate field")
```

Comment:
IN many cases, the error will be very hard to reason about without the `out`

Reply:
i am assuming you mean the cmd here.
Since the out would be irrelevant if the err != nil, right?